apteryx.c

```diff
@@ -275,39 +331,23 @@ handle_watch (rpc_message msg)
         return true;
     }
 
+    root = break_up_trunk(rpc_msg_decode_tree (msg));
+    /* Replace NULL end nodes with "" */
+    g_node_traverse (root, G_IN_ORDER, G_TRAVERSE_LEAVES, -1, replace_null_with_empty_string, NULL);
```

Comment:
is this still needed ... or was it a fix for lua needing an empty string ... I am worried this changes the behaviour a bit?

Reply:
The previous code had this: 

       if (value[0] == '\0')
                value = NULL;

which did the equivalent